packit/local_project.py

```diff
@@ -452,12 +476,53 @@ def checkout_pr(self, pr_id: Union[str, int]):
             "merge-requests" if is_gitlab else "pull", pr_id
         )
         remote_name = self.remote or "origin"
-        local_ref = f"refs/remotes/{remote_name}/pr/{pr_id}"
-        local_branch = f"pr/{pr_id}"
-        self.git_repo.remotes[remote_name].fetch(f"{remote_ref}:{local_ref}")
-        self.git_repo.create_head(local_branch, f"{remote_name}/{local_branch}")
-        self.git_repo.branches[local_branch].checkout()
-        logger.info(f"Checked out commit {self.git_repo.head.commit}")
+        local_ref = f"refs/remotes/{remote_name}/pr-changes/{pr_id}"
+        local_branch = f"pr-changes/{pr_id}"
+
+        self._fetch_as_branch(remote_ref, local_ref, local_branch)
+
+        head_commit = self.git_repo.branches[local_branch].commit
+        logger.info(
+            f"Checked out commit\n"
+            f"({shorten_commit_hash(head_commit.hexsha)})\t{head_commit.summary}"
+        )
+
+    def merge_pr(self, pr_id: Union[str, int]) -> None:
+        """
+        Merge given PR into target branch. Fetches and switches to base branch
+        (where changes from the PR are to be merged) and then merges branch with
+        changes from the PR.
+
+        Args:
+            pr_id: ID of the PR we are merging.
+
+        Raises:
+            PackitException: In case merge fails.
+        """
+        remote = self.remote or "origin"
+        pr = self.git_project.get_pr(int(pr_id))
+
+        local_target_branch = f"pr/{pr_id}"
```

Comment:
I am still confused about what the `target` means..;) Can you please pick a better name so it's clear? /Sadly, this is a more general problem and there is an issue for it somewhere../

Reply:
Yeah, it is… Especially since GitHub and GitLab have different understanding of `base`, it's mentioned on L492:

> Merge given PR into target branch.

Which is in my opinion descriptive, since there is both _merge into_ and _target branch_ :man_shrugging: